### PR TITLE
Added support for method Nedb providers

### PR DESCRIPTION
**Pull request by naranja0213**
npm start working ...
postman test working...
but still reminder error code: ELIFECYCLE..

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,7 @@
 node_modules/
-.devcontainer
\ No newline at end of file
+.devcontainer
+package-lock.json
```
Comment (CAVA1611) [inline .gitignore]: This file must not be in the .gitignore

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,7 @@
...
+package.json
```
Comment (CAVA1611) [inline .gitignore]: This file must not be in the .gitignore

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,7 @@
...
+index.js
```
Comment (CAVA1611) [inline .gitignore]: This file must not be in the .gitignore

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,7 @@
...
+.gitignore
```
Comment (CAVA1611) [inline .gitignore]: This file must not be in the .gitignore

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,7 @@
...
+index.js
```
Comment (CAVA1611) [inline .gitignore]: This file must not be in the .gitignore

File: index.js [hunk 6]
```diff
@@ -1,8 +1,10 @@
 var express = require('express');
 var bodyParser = require('body-parser');
+var DataStore = require("nedb");
 
 var port = 4500;
 var BASE_API_PATH = "/api/v1";
+var DB_FILE_NAME = __dirname+"/provider-v1.json";
 
 var providers = [
     {"cif": "A58818501", "name": "Proveedor 1", "address": "Sevilla", "cp":"20011", "phone": "0976543234", "email":"provedor1@gmail.com"},
```
Comment (CAVA1611) [inline index.js]: Se debería borrar providers, ya que se empezó a usar nedb.

File: index.js [hunk 7]
```diff
@@ -27,6 +35,15 @@ app.get(BASE_API_PATH + "/providers", (req, res) =>{
 app.post(BASE_API_PATH + "/providers", (req, res) => {
     console.log(Date() + "- POST /providers");
     var provider = req.body;
+    db.insert(providers,(err)=>{
```
Comment (CAVA1611) [inline index.js]: The correct variable to insert the document must be "provider".

File: index.js [hunk 8]
```diff
@@ -29,7 +29,16 @@ app.get("/", (req, res) => {
 
 app.get(BASE_API_PATH + "/providers", (req, res) =>{
     console.log(Date() + "- GET /providers");
+    db.find ({},(err,providers) =>{
+        if (err) {
+            console.log(Date() + "-" + err );
+            res.sendStatus(500);
+        }else {
+            res.sendStatus(providers);
```
Comment (CAVA1611) [inline index.js]: I think the correct command is only "send" not "sendStatus"

**Review by CAVA1611 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (naranja0213) [reply .gitignore]: done

File: .gitignore (hunk 2 above)
Comment (naranja0213) [reply .gitignore]: done

File: .gitignore (hunk 3 above)
Comment (naranja0213) [reply .gitignore]: done

File: .gitignore (hunk 4 above)
Comment (naranja0213) [reply .gitignore]: done

File: .gitignore (hunk 5 above)
Comment (naranja0213) [reply .gitignore]: done

File: index.js (hunk 6 above)
Comment (naranja0213) [reply index.js]: done ,ya hecho

File: index.js (hunk 7 above)
Comment (naranja0213) [reply index.js]: ya hecho

File: index.js (hunk 8 above)
Comment (naranja0213) [reply index.js]: done

**Review by CAVA1611 — APPROVED** [review]

**Comment by CAVA1611** [comment]
ok

**Pull request closed without merging**